pkg/api/store/hpa/store.go

```diff
@@ -0,0 +1,46 @@
+package hpa
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/types"
+	mgmtv3 "github.com/rancher/types/apis/management.cattle.io/v3"
+)
+
+type NoWatchByClusterVersionStore struct {
+	types.Store
+	ClusterLister mgmtv3.ClusterLister
+}
+
+func (s *NoWatchByClusterVersionStore) Watch(apiContext *types.APIContext, schema *types.Schema, opt *types.QueryOptions) (chan map[string]interface{}, error) {
+	clusterName := getClusterName(apiContext)
+	obj, err := s.ClusterLister.Get("", clusterName)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get cluster obj from lister")
+	}
+	version := obj.Status.Version
+	if version == nil {
+		return nil, nil
+	}
+
+	// autoscaing v2beta2 was released in v1.12 https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.12.md#sig-autoscaling
+	if version.Minor < "12" {
+		return make(chan map[string]interface{}), nil
+	}
+	return s.Store.Watch(apiContext, schema, opt)
+}
+
+func getClusterName(apiContext *types.APIContext) string {
+	clusterID := apiContext.SubContext["/v3/schemas/cluster"]
+	if clusterID == "" {
+		projectID, ok := apiContext.SubContext["/v3/schemas/project"]
+		if ok {
+			parts := strings.SplitN(projectID, ":", 2)
```

Comment:
Can we do `clusterID, _ = ref.Parse(projectID)` here

Reply:
Updated

Reply:
I am reusing the `ClusterName` function from `clustermanager.Manager`.